src/onnx/onnx.cpp

```diff
@@ -441,11 +458,18 @@ struct onnx_parser
         if(auto_pad.find("SAME") != std::string::npos)
         {
             bool is_same_upper = (auto_pad.find("SAME_UPPER") != std::string::npos);
-            std::vector<int64_t> padding(in_lens.size());
-            calculate_padding(
-                0, padding, in_lens[2], op.stride[0], dilation[0], k_lens[0], is_same_upper);
-            calculate_padding(
-                1, padding, in_lens[3], op.stride[1], dilation[1], k_lens[1], is_same_upper);
+            std::vector<int64_t> padding(2 * (in_lens.size() - 2));
+
+            for(size_t i = 0; i < padding.size() / 2; i++)
+            {
+                calculate_padding(i,
```

Comment:
The calculate_padding function needs to be changed as:
auto pad_ndim = pads.size() / 2;
pads[idx] = pad/2;
pads[idx + pad_ndim] = pad - pad / 2;